### PR TITLE
Default Config for Staking

### PR DESCRIPTION
**Pull request by Ank4n**

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 3/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3720675

File: substrate/frame/babe/src/mock.rs [hunk 1]
```diff
@@ -63,30 +63,17 @@ frame_support::construct_runtime!(
 	}
 );
 
+#[derive_impl(frame_system::config_preludes::TestDefaultConfig as frame_system::DefaultConfig)]
 impl frame_system::Config for Test {
 	type BaseCallFilter = frame_support::traits::Everything;
-	type BlockWeights = ();
-	type BlockLength = ();
-	type DbWeight = ();
 	type RuntimeOrigin = RuntimeOrigin;
```
Comment (gupnik) [inline substrate/frame/babe/src/mock.rs:69]: `RuntimeOrigin`, `RuntimeCall` and `RuntimeEvent` can also be removed now.

File: substrate/frame/staking/src/pallet/mod.rs [hunk 2]
```diff
@@ -82,9 +82,36 @@ pub mod pallet {
...
+			type MaxUnlockingChunks = ConstU32<32>;
+			type HistoryDepth = ConstU32<84>;
+			type WeightInfo = ();
+		}
+	}
+
+	#[pallet::config(with_default)]
 	pub trait Config: frame_system::Config {
 		/// The staking balance.
+		#[pallet::no_default]
```
Comment (gupnik) [inline substrate/frame/staking/src/pallet/mod.rs:114]: We can try using `no_default_bounds` for a number of them.

File: substrate/frame/staking/src/pallet/mod.rs [hunk 3]
```diff
@@ -82,9 +82,36 @@ pub mod pallet {
 		Remove,
 	}
 
-	#[pallet::config]
+	/// Some default provided implementations of [`DefaultConfig`], which can be used to implement
```
Comment (kianenigma) [inline substrate/frame/staking/src/pallet/mod.rs:85]: This can be then applies in a couple of pallets, like `grandpa` etc?

**Review by kianenigma — COMMENTED** [review]
Thanks! 

Please update #171 once done.

**Pull request closed without merging**